evaluator/evaluate.go

```diff
@@ -158,27 +174,86 @@ func evaluateWithVmaas(ctx context.Context, tx *gorm.DB, updatesReq *vmaas.Updat
 		evaluationCnt.WithLabelValues("error-database-commit").Inc()
 		return nil, errors.New("database commit failed")
 	}
-	return vmaasData, nil
+	return updatesData, nil
 }
 
-func tryGetVmaasRequest(tx *gorm.DB, accountID int, inventoryID string,
-	requested *base.Rfc3339Timestamp) (*vmaas.UpdatesV3Request, *models.SystemPlatform, error) {
-	system := tryGetSystem(tx, accountID, inventoryID, requested)
-	if system == nil {
-		return nil, nil, nil
+func getUpdatesData(ctx context.Context, tx *gorm.DB, system *models.SystemPlatform) (*vmaas.UpdatesV2Response, error) {
+	if !enableYumUpdatesEval {
+		return getVmaasUpdates(ctx, tx, system)
+	}
+
+	var vmaasErr error
+	var yumErr error
+	vmaasData, vmaasErr := getVmaasUpdates(ctx, tx, system)
+	yumUpdates, yumErr := tryGetYumUpdates(system)
+	if vmaasErr != nil && yumErr != nil {
+		return nil, errors.Wrap(vmaasErr, yumErr.Error())
+	}
+	if vmaasErr != nil || vmaasData == nil {
+		return yumUpdates, nil
+	}
+	if yumErr != nil || yumUpdates == nil {
+		return vmaasData, nil
 	}
+	if yumUpdates == nil && vmaasData == nil {
+		return nil, nil
+	}
+
+	// Return YumUpdates if it is uploader
+	if evalTopic == uploadTopic {
+		return yumUpdates, nil
+	}
```

Comment:
This `if` should go before `getVmaasUpdates()` otherwise you unnecessarily call (and wait for and parse) vmaas and then just throw the response away.